pil-analyzer/src/pil_analyzer.rs

```diff
@@ -174,6 +176,99 @@ impl PILAnalyzer {
         }
     }
 
+    /// Check that all match expressions are exhaustive.
+    pub fn match_exhaustiveness_check(&self) {
+        for (name, (_, value)) in &self.definitions {
+            let Some(value) = value else { continue };
+            for e in value.all_children() {
+                let patterns = if let Expression::MatchExpression(_, arms) = e {
+                    arms.iter()
+                        .map(|arm| PatternTuple {
+                            patterns: vec![arm.pattern.clone()],
+                        })
+                        .collect::<Vec<_>>()
+                } else {
+                    continue;
+                };
+                let witnesses = Self::usefulness(&patterns, Pattern::CatchAll.into());
+                if witnesses.is_empty() {
+                    panic!("Function {name} is not exhaustive")
+                }
+            }
+        }
+
+        for id in &self.identities {
+            let patterns = id
+                .all_children()
+                .filter_map(|e| {
```

Comment:
Can you try to re-use the code from above? Maybe there is a way to use an iterator through both structures.